core/src/avm1/globals/system.rs

```diff
@@ -0,0 +1,544 @@
+            .append_pair(
+                "M",
+                &self.encode_string(
+                    self.manufacturer
+                        .get_manufacturer_string(avm.player_version)
+                        .as_str(),
+                ),
+            )
+            .append_pair(
+                "R",
+                &format!("{}x{}", self.screen_resolution.0, self.screen_resolution.1),
+            )
+            .append_pair("COL", self.screen_color.get_color_code())
+            .append_pair("AR", &self.aspect_ratio.to_string())
+            .append_pair("OS", &self.encode_string(self.os.get_os_name()))
+            .append_pair("L", self.language.get_language_code(avm.player_version))
+            .append_pair("IME", self.encode_capability(SystemCapabilities::IME))
+            .append_pair("PT", self.player_type.get_player_name())
+            .append_pair(
+                "AVD",
+                self.encode_not_capability(SystemCapabilities::AvHardware),
+            )
+            .append_pair(
+                "LFD",
+                self.encode_not_capability(SystemCapabilities::LocalFileRead),
+            )
+            .append_pair("DP", &self.dpi.to_string())
+            .finish()
+    }
+}
+
+impl Default for SystemProperties {
+    fn default() -> Self {
+        SystemProperties {
+            //TODO: default to true on fp>=7, false <= 6
+            exact_settings: true,
+            //TODO: default to false on fp>=7, true <= 6
+            use_codepage: false,
+            capabilities: EnumSet::empty(),
+            player_type: PlayerType::StandAlone,
+            screen_color: ScreenColor::Color,
+            // TODO: note for fp <7 this should be the locale and the ui lang for >= 7, on windows
+            language: Language::English,
+            screen_resolution: (0, 0),
+            aspect_ratio: 1_f32,
+            dpi: 1_f32,
+            manufacturer: Manufacturer::Linux,
+            os: OperatingSystem::Linux,
+            sandbox_type: SandboxType::LocalTrusted,
+        }
+    }
+}
+
+pub fn set_clipboard<'gc>(
+    avm: &mut Avm1<'gc>,
+    action_context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let new_content = args
+        .get(0)
+        .unwrap_or(&Value::Undefined)
+        .to_owned()
+        .coerce_to_string(avm, action_context)?;
+
+    action_context.input.set_clipboard_content(new_content);
+
+    Ok(Value::Undefined.into())
+}
+
+pub fn show_settings<'gc>(
+    _avm: &mut Avm1<'gc>,
+    _action_context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    //TODO: should default to the last panel displayed
+    let last_panel = SettingsPanel::Privacy;
+
+    let panel = args
+        .get(0)
+        .map(|v| match v {
+            Value::Number(x) => SettingsPanel::try_from(*x as u8).unwrap_or(last_panel),
```

Comment:
Need to do something like `v.coerce_to_i32(avm, context)?` for the weird cases such as:
```
System.showSettings(
  { valueOf: function(){ return 2; } }
);
```